### DISCUSSION TITLE
I don't really get how this should work

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [nordeck/jitsi-keycloak-adapter#26] [General]
I am not even getting this to work if I test with this url

`/static/oidc-redirect.html`

I am redirected to:

`/static/oidc-redirect.html?oidc=unauthorized`

and get a forbidden

Why not a link in these popups?
![2](https://github.com/user-attachments/assets/cfdf17db-9e3f-4891-a67c-24fe644239e9)
![3](https://github.com/user-attachments/assets/778a1147-0e6d-4505-9a01-0bf4224aa0c6)

Like eg most wordpress plugins do

![1](https://github.com/user-attachments/assets/c4d96362-953d-4beb-9113-f25384d62fa5)

I am quite curious how this works, because now guests get these dialogs and wait for the organizer to login. Is this all going to stay the same? From this manual[1] I don't really get how this maintains the current functionality. 

https://github.com/nordeck/jitsi-keycloak-adapter/blob/main/docs/setup-standalone.md

**Comment by emrahcom** [comment]
Could you share your `/usr/share/jitsi-meet/body.html`?

**Reply by f1-outsourcing** [reply]
Looks like it is only javascript, but this one has a new date, so probably from testing with sso

```
<script>
  function oidcRedirect() {
    const qs = new URLSearchParams(window.location.search.substring(1));
    qs.delete("oidc");

    const path = encodeURIComponent(
      window.location.pathname.replace(/\/+/g, "/"),
    );
    const search = encodeURIComponent(qs.toString());

    let hash = "config.prejoinConfig.enabled=false";
    if (window.location.hash) {
      hash = `${hash}&${window.location.hash.substring(1)}`;
    }
    hash = encodeURIComponent(hash);

    window.location = `/oidc/auth?path=${path}&search=${search}&hash=${hash}`;

    try {
      // remove react from DOM to prevent UI distortion
      document.all.react.remove();
    } catch {
      // do nothing
    }
  }

  function interceptLoginRequest() {
    try {
      // check if login or IamHost button is created in DOM
      const _button = document.getElementById("modal-dialog-ok-button");
      if (!_button) return;

      let labelKey;
      try {
        labelKey = Object.values(_button)[0].return.memoizedProps.labelKey;
      } catch {
        // do nothing
      }

      if (labelKey === "dialog.login") {
        // if this is a login screen, redirect to OIDC login page
        oidcRedirect();
      } else if (labelKey === "dialog.IamHost") {
        // if this is an IamHost screen, redirect to OIDC login page when clicked
        _button.onclick = oidcRedirect;
      }
    } finally {
      setTimeout(function () {
        interceptLoginRequest();
      }, 1000);
    }
  }

  interceptLoginRequest();
</script>
```

**Comment by emrahcom** [comment]
The moderator should click the "I am the host" button to go to SSO page if she is not authenticated yet.

By the way, is this an old deployment? This popup is different for the latest releases.

**Reply by f1-outsourcing** [reply]
> The moderator should click the "I am the host" button to go to SSO page if she is not authenticated yet.

Ok but it is not possible to keep the regular login and have a SSO login link?

> By the way, is this an old deployment? This popup is different for the latest releases.

Yes I had issue with upgrading, currently stuck

**Reply by emrahcom** [reply]
Jitsi doesn't need the authentication at this stage. It will goes to SSO page when it is needed (_after clicking the button_).
